tee-worker/litentry/core/credentials/src/credentials.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+use crate::error::Error;
+use chrono::{DateTime, FixedOffset, NaiveDateTime, SecondsFormat};
+use codec::{Decode, Encode};
+use itp_types::AccountId;
+use itp_utils::stringify::account_id_to_string;
+use litentry_primitives::Assertion;
+use log::*;
+use scale_info::TypeInfo;
+use serde::{Deserialize, Serialize};
+use std::{
+	fmt::Debug,
+	string::{String, ToString},
+	time::{SystemTime, UNIX_EPOCH},
+	vec::Vec,
+};
+
+pub const PROOF_PURPOSE: &str = "assertionMethod";
+pub const MAX_CREDENTIAL_SIZE: usize = 2048;
+
+pub fn now() -> String {
+	let ts = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
```

Comment:
Will this part of code execute inside TEE? I think "SystemTime::now()" is suspicious to run inside TEE.

Reply:
- It is not possible to have a trusted and precise timestamp inside TEE, but it is feasible to measure the **Duration** between two timestamps from the support of teaclave sgx_tstd https://github.com/apache/incubator-teaclave-sgx-sdk/blob/master/sgx_tstd/src/time.rs

- The "SystemTime::now()" could run inside TEE enclave, the scenario is here https://github.com/litentry/litentry-parachain/blob/250d02a9affddaeda4d629283a35006bc3841387/tee-worker/enclave-runtime/src/top_pool_execution.rs#L94